src/processors/ipfsFile/mediaUploader.ts

```diff
@@ -0,0 +1,96 @@
+import { urlSource } from '../../clients/ipfs';
+import { Table } from '../../db/db';
+import { IPFSFile, IPFSFileUrl } from '../../types/ipfsFile';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+type TrackUrlJoin = ProcessedTrack & Partial<IPFSFileUrl>;
+type MediaType = 'Artwork' | 'Audio';
+
+const notOnIpfs: (media: MediaType) => Trigger<undefined> =
+(media) => async (clients: Clients) => {
+  const query = `select * from "${Table.processedTracks}" as t
+      left outer join "${Table.ipfsFilesUrls}" as fileUrl
+      on t."lossy${media}URL" = fileUrl.url
+      where "lossy${media}IPFSHash" is null
+      and "lossy${media}URL" is not null
+      and fileUrl.error is null
+      LIMIT ${process.env.IPFS_UPLOAD_BATCH_SIZE || process.env.QUERY_TRIGGER_BATCH_SIZE!}`
+
+  const trackUrlJoin = (await clients.db.rawSQL(
+    query
+  )).rows
+
+  return trackUrlJoin
+};
+
+function processorFunction(media: MediaType) {
+  return async (tracksWithIPFSFiles: TrackUrlJoin[], clients: Clients) => {
+    const trackUpdates: Partial<ProcessedTrack>[] = [];
+    const ipfsFiles: IPFSFile[] = [];
+    const ipfsFilesUrls: IPFSFileUrl[] = [];
+
+    const urlField = `lossy${media}URL`
+    const ipfsField = `lossy${media}IPFSHash`
+
+    const processTrack = async (trackWithUrl: TrackUrlJoin) => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const url = trackWithUrl[urlField]
+      const source = urlSource(url)
+      let cid = trackWithUrl['cid']
+
+      delete trackWithUrl['url']
+      delete trackWithUrl['cid']
+      delete trackWithUrl['error']
+      delete trackWithUrl['lastRetry']
+      delete trackWithUrl['numberOfRetries']
+
+      const track: ProcessedTrack = {
+        ...trackWithUrl
+      }
```

Comment:
Instead of creating this track update with all the fields and then deleting fields we don't need, can we just create the record update using only the exact fields we need (ie, id(s) and cidField only)?

The deletion flow is not future-compatible, since it basically assumes that once the extra fields are deleted, the update remaining is minimal, but if we change/add fields in future, this could break. An exact-field creation flow should not introduce this implicit assumption/risk.